docker/tpu/Dockerfile.cluster

```diff
@@ -0,0 +1,74 @@
+# This dockerfile is used to build the docker image for using Ray to manage TPU slices.
+ARG IMAGE=ghcr.io/stanford-crfm/levanter-base
+ARG TAG=latest
+
+FROM ${IMAGE}:${TAG}
+
+# install docker in docker, but don't start it
+RUN apt-get update && apt-get install -y docker.io
+
+ENV TENSORSTORE_CURL_LOW_SPEED_TIME_SECONDS=60\
+    TENSORSTORE_CURL_LOW_SPEED_LIMIT_BYTES=1024\
+    RAY_USAGE_STATS_ENABLED=0\
+    PATH=/opt/levanter/.venv/bin:$PATH\
+    PYTHONPATH=/opt/levanter:/opt/levanter/src:/opt/levanter/examples:/opt/levanter/tests:src:.\
+    HOME=/home/levanter
+# Install dependencies
+
+RUN apt-get install -y \
+        sudo \
+        git \
+        libjemalloc-dev \
+        wget \
+        cmake \
+        g++ \
+        zlib1g-dev \
+        tmux \
+        screen \
+        rsync \
+        netbase \
+        openssh-client \
+        gnupg
+
+RUN pip install --no-cache-dir \
+        flatbuffers \
+        cython==0.29.37 \
+        # Necessary for Dataset to work properly.
+        numpy\>=1.20 \
+        psutil \
+        # Required a recent version of setuptools to be compatible with python 3.12+.
+        setuptools==71.1.0 \
+          "google-api-python-client==1.7.8" \
+        "google-oauth"
+
+
+RUN curl https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz > /tmp/google-cloud-sdk.tar.gz
+
+# Installing the package
```

Comment:
Maybe a comment for why we're installing this into the image:

# gcloud CLI is used during setup to fetch secrets and configure docker access to GHCR

Reply:
good idea